### PR TITLE
Add warning when `wandb.run` already exists

### LINKED ISSUE #8614: 5-Fold with PyTorchLightning + Wandb seems to log to the same experiment
I am training 5-fold CV with PyTorch Lightning in a for loop. I am also logging all the results to wandb. I want wanbd to reinitalize the run after each fold, but it seems to continue with the same run and it logs all the results to the same run. I also tried passing kwargs in the WandbLogger as mentioned in the docs [here](https://pytorch-lightning.readthedocs.io/en/stable/extensions/generated/pytorch_lightning.loggers.WandbLogger.html#pytorch_lightning.loggers.WandbLogger.params.**kwargs), with no luck.
Here's a pseudo code of it:

```python
def run(fold):
    kwargs = {
        "reinit": True,
        "group": f"{CFG['exp_name']}"
    }
    wandb_logger = WandbLogger(project='<name>', 
                        entity='<entity>', 
                        config = CFG,
                        name=f"fold_{fold}",
                        **kwargs
            )
    trainer = Trainer(
        precision=16,
        gpus=1,
        fast_dev_run=False,
        callbacks = [checkpoint_callback],
        logger=wandb_logger,
        progress_bar_refresh_rate=1,
        max_epochs=2,
        log_every_n_steps=1

    )

    trainer.fit(
        lit_model,
        data_module
    )

if __name__ == "__main__":
    for fold in range(5):
        run(fold)
```

_Originally posted by @Gladiator07 in https://github.com/PyTorchLightning/pytorch-lightning/discussions/8572_

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #8614

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Comment by borisdayma** [comment]
LGTM 🥳

File: pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -190,7 +190,14 @@ def experiment(self) -> Run:
         if self._experiment is None:
             if self._offline:
                 os.environ["WANDB_MODE"] = "dryrun"
-            self._experiment = wandb.init(**self._wandb_init) if wandb.run is None else wandb.run
+            if wandb.run is None:
+                self._experiment = wandb.init(**self._wandb_init)
+            else:
+                warning_cache.warn(
```
Comment (carmocca) [inline pytorch_lightning/loggers/wandb.py:196]: Why do we need the warning cache if this will only runk on a single rank?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/loggers/wandb.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/loggers/wandb.py:196]: warning cache is already used in a few places here in this logger. 
probably for jupyter notebook when they re-run the cell it would not show the warning multiple times. not sure.

Comment (carmocca) [reply pytorch_lightning/loggers/wandb.py:196]: > probably for jupyter notebook when they re-run the cell it would not show the warning multiple times

wouldn't you want to still show the warning if re-running the cell?

> already used in a few places here in this logger

it was initially introduced in #5931. @borisdayma did you have a specific reason?

Comment (borisdayma) [reply pytorch_lightning/loggers/wandb.py:196]: I think there are no more reasons for it.
It was used when we were dropping values in the past due to `global_step` (not an issue anymore).
Then I kept it for the rest mainly t o have the right formatting of display (and also because it was already there).

I agree that you would want to show the warning again in notebooks.

Comment (awaelchli) [reply pytorch_lightning/loggers/wandb.py:196]: see #8764

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by justusschock — APPROVED** [review]
